.github/workflows/run-acceptance-tests.yml

```diff
@@ -439,6 +441,97 @@ jobs:
             name: Upload coverage reports to Codecov
             uses: codecov/codecov-action@v3
       timeout-minutes: 60
+  test_oidc:
+      if: github.event_name == 'repository_dispatch' || github.event.pull_request.head.repo.full_name == github.repository
+      name: test_oidc
+      needs: build_sdk
+      permissions:
+          contents: read
+          id-token: write
+      runs-on: pulumi-ubuntu-8core
+      steps:
+          - name: Checkout Repo
+            uses: actions/checkout@v3
+            with:
+              ref: ${{ env.PR_COMMIT_SHA }}
+              submodules: true
+          - name: Checkout Scripts Repo
+            uses: actions/checkout@v3
+            with:
+              path: ci-scripts
+              repository: pulumi/scripts
+          - name: Unshallow clone for tags
+            run: git fetch --prune --unshallow --tags
+          - name: Install Go
+            uses: actions/setup-go@v4
+            with:
+              cache-dependency-path: |
+                  sdk/go.sum
+              go-version: 1.21.x
+          - name: Install pulumictl
+            uses: jaxxstorm/action-install-gh-release@v1.5.0
+            with:
+              repo: pulumi/pulumictl
+          - name: Install Pulumi CLI
+            uses: pulumi/actions@v4
+            with:
+              pulumi-version: v3.77.1
+          - name: Setup Node
+            uses: actions/setup-node@v2
+            with:
+              node-version: ${{ env.NODEVERSION }}
+              registry-url: https://registry.npmjs.org
+          - name: Download provider + tfgen binaries
+            uses: actions/download-artifact@v2
+            with:
+              name: ${{ env.PROVIDER }}-provider.tar.gz
+              path: ${{ github.workspace }}/bin
+          - name: Untar provider binaries
+            run: |-
+              tar -zxf ${{ github.workspace }}/bin/provider.tar.gz -C ${{ github.workspace}}/bin
+              find ${{ github.workspace }} -name "pulumi-*-${{ env.PROVIDER }}" -print -exec chmod +x {} \;
+          - name: Download SDK
+            uses: actions/download-artifact@v2
+            with:
+              name: ${{ matrix.language }}-sdk.tar.gz
+              path: ${{ github.workspace}}/sdk/
+          - name: Uncompress SDK folder
+            run: tar -zxf ${{ github.workspace }}/sdk/${{ matrix.language }}.tar.gz -C ${{ github.workspace }}/sdk/${{ matrix.language }}
+          - name: Update path
+            run: echo "${{ github.workspace }}/bin" >> "$GITHUB_PATH"
+          - name: Install dependencies
+            run: make install_${{ matrix.language}}_sdk
+          - name: Install gotestfmt
+            uses: GoTestTools/gotestfmt-action@v2
+            with:
+              token: ${{ secrets.GITHUB_TOKEN }}
+              version: v2.4.0
+          - name: Make upstream
+            run: make upstream
+          - name: Run selected tests with manual web identity/OIDC auth
```

Comment:
So this works I guess but do we have any alternatives that don't create a brand new test target for this particular test to keep the workflows a little simper? 

What are the extra requirements for the test? I see you need OIDC_ROLE_ARN but perhaps this could have been exposed to all tests? Then you got `aws-actions/configure-aws-credentials@v4` that's assuming the role with `role-to-assume`. Is that the only way to do it? 

Can Pulumi program under test assume this role, perhaps by using explicit provider configuration?

If it can't, can the test harness take care of it in the Go code?

Reply:
I made a copy of the `test` job because it runs `aws-actions/configure-aws-credentials` and I want to run it, too, but with a different configuration. If I could edit this file directly, I could just append the steps to the existing `test` job but in ci-mgmt, I can't.

This new OIDC test job runs two `go test` runs:
1. with manual OIDC configuration
2. with `aws-actions/configure-aws-credentials`

I believe I could port (1) to the existing test job. Since (1) proves that OIDC works, (2) mainly tests the OIDC part of `aws-actions/configure-aws-credentials` which we don't own. So I guess we could just drop it, although I would have liked to have an e2e test for a common user scenario.

Reply:
Yup. It makes sense to have an e2e test for a common user scenario.